chapter1/03_Memory_organization.rst

```diff
@@ -1,2 +1,95 @@
 内存管理
 ========
+
+内存管理是并行架构需要考虑的另一方面，确切来说是获得数据的方式。无论处理单元多快，如果内存提供指令和数据的速度跟不上，就会造成系统的瓶颈。制约内存响应时间与处理器速度相匹配的主要因素是内存循环时间。所谓循环时间就是两次连续操作之间所消耗的时间。处理器的循环时间通常比内存循环时间短得多。当处理器传送数据到内存或从内存中获取数据时，内存依旧在内存循环中，其他任何设备（I/O控制器，处理器）都不能使用内存，以为内存必须先对上一个请求作出响应。
+
+.. image:: ../images/Page-7-Image-1.png
+
+不同的人为了解决 MIMD 架构获取内存数据的问题，分别提出了两种内存管理系统。第一种就是人们所熟知的共享内存系统，共享内存系统有大量的虚拟内存空间，而且处理器能够公平访问内存中的数据和指令。另外一种类型是分布式内存模型，在这种内存模型中，每个处理器都有自己专属的内存，其他处理器都不能访问。共享内存和分布式内存本质上的区别是虚拟内存的结构，以处理器的角度来说是内存的位置不一样。 一般来说，每个系统的内存都会分为能独立访问的不同部分，共享内存和分布式内存系统的处理单元管理内存访问的方式不同。 
+``load R0,i`` 
+指令意味着将内存中
+``i``
+位置的内容加载进
+``R0``
+寄存器，但在不同的内存管理下处理器的处理方式不尽相同。在共享内存的系统中，
+``i``
+代表的是内存的全局地址，对于系统中的所有处理器都指向同一块内存空间。如果两个处理器通向想要执行该内存中的指令，它们会向
+``R0``
+寄存器载入相同的信息。在分布式内存系统中，
+``i``
+是局部地址。如果两个处理器同时执行向
+``R0``
+载入内容的语句，执行结束之后，不同处理器
+``R0``
+寄存器中的值一般情况下是不一样的，因为每个处理器对应的内存块中的``i``代表的全局地址不一样。对于程序员来说，必须区分共享内存和分布式内存，因为在并行编程中需要考量内存管理方式来决定进程或线程间通信的方式。对于系统来说，共享内存能够在内存中构建数据结构并在子进程间直接交换该数据结构的引用。而对于分布式内存系统来说，必须在局部内存保存共享数据的副本。一个处理器会向其他处理器发送含有共享数据的消息从而创建数据副本。这就导致了这种内存管理的一个缺点，那就是，当要发送的消息太大，发送过程会耗费相对较长的时间。
+
+共享内存
+--------
+
+下图展示了共享内存多处理器系统的架构，这里只展示了各部件之间的简单连接。总线结构允许任意数量的设备共享通道。总线协议最初设计是让单处理器，一个或多个磁盘和磁带控制器通过共享内存进行通信。各个处理器共用一个Cache，Cache中保存着局部内存中很有可能被处理器使用的指令或数据。可以想象以下，当一个处理器修改了内存中的数据，同时另外一个处理器正在使用这个数据时，就会出现问题。已修改的值会从处理器的Cache传递到共享内存中，接着，这个新值对所有处理器都是可见的，那么它们希望使用旧值完成运算是不可能的。这就是人们所熟知的Cache一致性问题，内存一致性问题的特殊情况，要解决这个问题需要硬件像多进程编程一样实现处理并发问题和同步控制。
```

Comment:
Note that each processor has been associated with a cache memory, as it is assumed that the probability that a processor needs data or instructions present in the local memory is very high.

各个处理器共用一个Cache

这个地方不太对，应该是每个处理器分配了一个Cache

Reply:
已修改的值会从处理器的Cache传递到共享内存中，接着，这个新值对所有处理器都是可见的，那么它们希望使用旧值完成运算是不可能的。

 The new value will pass from the processor cache that has been changed to shared memory; later, however, it must also be passed to all the other processors, so that they do not work with the obsolete value. This problem is known as the problem of cache coherency, a special case of the problem of memory consistency, which requires hardware implementations that can handle concurrency issues and synchronization similar to those having thread programming.

---

不太恰当，应该说

> 接着，新值必须被传到其他的Cache，这样才能保证其他处理器不会使用旧值计算。

Reply:
1. >每个处理器分配一个Cache

    我改成了

    >处理器拥有各自的 Cache。

    一开始没有留意示意图
2. 一开始按照类似读“脏”数据来处理，翻译起来有些偏颇。

Reply:
>接着，新值必须被传到其他的Cache，这样才能保证其他处理器不会使用旧值计算。

这里我翻译成了

>当一个处理器修改了内存中的数据，同时另外一个处理器正在使用这个数据时，就会出现问题。已修改的值会从处理器的Cache传递到共享内存中，接着，新值会传递到其他处理器的Cache中，其它处理器就不可以使用旧值进行计算。

我觉的在这句话的语境下，应该是提出问题，如果按照你的那种翻译，就像是解决了问题一样